test/utils/MessagesSpec.scala

```diff
@@ -0,0 +1,190 @@
+    "niphlQuestion.p2",
+    "niphlQuestion.p2.linkText",
+    "niphlQuestion.h2",
+    "niphlQuestion.radio.notSelected",
+    "ukimsNumber.title",
+    "ukimsNumber.h1",
+    "ukimsNumber.p1",
+    "ukimsNumber.p2",
+    "ukimsNumber.heading",
+    "ukimsNumber.hint",
+    "ukimsNumber.linkText",
+    "ukimsNumber.error.required",
+    "ukimsNumber.error.invalidFormat",
+    "nirmsQuestion.title",
+    "nirmsQuestion.h1",
+    "nirmsQuestion.p1",
+    "nirmsQuestion.p2",
+    "nirmsQuestion.p2.linkText",
+    "nirmsQuestion.h2",
+    "nirmsQuestion.error.notSelected",
+    "nirmsNumber.title",
+    "nirmsNumber.heading",
+    "nirmsNumber.hint",
+    "nirmsNumber.error.required",
+    "nirmsNumber.error.invalidFormat",
+    "niphlNumber.title",
+    "niphlNumber.h1",
+    "niphlNumber.p1",
+    "niphlNumber.li1",
+    "niphlNumber.li2",
+    "niphlNumber.li3",
+    "niphlNumber.h2",
+    "niphlNumber.hint",
+    "niphlNumber.error.notSupplied",
+    "niphlNumber.error.wrongFormat",
+    "checkYourAnswers.title",
+    "checkYourAnswers.heading",
+    "journeyRecovery.continue.title",
+    "journeyRecovery.continue.heading",
+    "journeyRecovery.continue.guidance",
+    "journeyRecovery.startAgain.title",
+    "journeyRecovery.startAgain.heading",
+    "journeyRecovery.startAgain.guidance"
+  )
+
+  "All message files" - {
+    "must have a non-empty message for each key" in {
+      assertNonEmptyNonTemporaryValues("en", englishMessages)
+    }
+    "must have no unescaped single quotes in value" in {
+      assertCorrectUseOfQuotes("en", englishMessages)
+    }
+    "must not have missing keys" in {
+      assertNoMissingKeys("en", englishMessages, serviceMessageKeys)
+    }
+
+  }
+
+  private def assertNonEmptyNonTemporaryValues(label: String, messages: Map[String, String]): Unit = messages.foreach {
+    case (key, value) =>
+      withClue(s"In $label, there is an empty value for the key:[$key][$value]") {
+        value.trim.nonEmpty mustBe true
+      }
+  }
+
+  private def assertCorrectUseOfQuotes(label: String, messages: Map[String, String]): Unit = messages.foreach {
+    case (key, value) =>
+      withClue(s"In $label, there is an unescaped or invalid quote:[$key][$value]") {
+        matchSingleQuoteOnly.findFirstIn(value).isEmpty mustBe true
+        matchBacktickQuoteOnly.findFirstIn(value).isEmpty mustBe true
+      }
+  }
+
+  private def assertNoMissingKeys(label: String, messages: Map[String, String], expectedKeys: Set[String]): Unit = {
+    val missingKeys = expectedKeys.diff(messages.keySet)
+    withClue(s"In $label, the following keys are missing: ${missingKeys.mkString(", ")}") {
+      missingKeys.isEmpty mustBe true
+    }
+  }
+
+  private lazy val englishMessages: Map[String, String] = getExpectedMessages("en") -- commonMessageKeys
+
+  private def getExpectedMessages(languageCode: String) =
+    messagesAPI.messages.getOrElse(languageCode, throw new Exception(s"Missing messages for $languageCode"))
```

Comment:
Instead of throwing an exception here you can return an Option, where if it's empty you fail the test:
Use Map[String, Option[String]], then value match { 
case Some(value) => ...
case None => fail("Missing messages for...")

Reply:
Thanks, updated now :)